packages/mui-system/src/sizing.js

```diff
@@ -15,7 +15,8 @@ export const maxWidth = (props) => {
   if (props.maxWidth !== undefined && props.maxWidth !== null) {
     const styleFromPropValue = (propValue) => {
       const breakpoint =
-        props.theme?.breakpoints?.values?.[propValue] || breakpointsValues[propValue];
+        `${props.theme?.breakpoints?.values?.[propValue]}${props.theme?.breakpoints?.unit || ''}` ||
```

Comment:
Another place where similar solution is applied: [here](https://github.com/mui/material-ui/blob/master/packages/mui-system/src/Container/createContainer.tsx#L104)